app/uk/gov/hmrc/hec/repos/HECTaxCheckStore.scala

```diff
@@ -93,7 +94,33 @@ class HECTaxCheckStoreImpl @Inject() (
       mongo.mongoConnector.db,
       ec
     ) {
-      override def indexes: Seq[Index] = hecIndexes
+
+      //TODO temporary till the issue is resolved by the team owning cacheRepository code
+      //issue- indexes were not getting created
+      @SuppressWarnings(Array("org.wartremover.warts.Throw"))
+      private def ensureIndex(index: Index)(implicit ec: ExecutionContext): Future[Boolean] =
+        collection.indexesManager
+          .create(index)
+          .map { wr =>
+            if (!wr.ok) {
+              val msg      = wr.writeErrors.mkString(", ")
+              val maybeMsg = if (msg.contains("E11000")) {
+                // this is for backwards compatibility to mongodb 2.6.x
+
+                throw GenericDatabaseException(msg, wr.code)
+              } else Some(msg)
+              logger.error(s"$message (${index.eventualName}) : '${maybeMsg.map(_.toString)}'")
+            }
+            wr.ok
+          }
+          .recover { case t =>
+            logger.error(s"$message (${index.eventualName})", t)
+            false
+          }
+
+      override def ensureIndexes(implicit ec: ExecutionContext): Future[Seq[Boolean]] =
```

Comment:
we should call `super.ensureIndexes` as well to make sure the TTLIndexing stuff is run